lib/_tls_common.js

```diff
@@ -153,6 +153,31 @@ exports.createSecureContext = function createSecureContext(options) {
     }
   }
 
+  const { privateKeyIdentifier, privateKeyEngine } = options;
+  if (privateKeyIdentifier !== undefined && privateKeyEngine !== undefined) {
+    if (key) {
+      // Both cleartext key and engine key can't be set at the same time
+      throw new ERR_INVALID_OPT_VALUE('privateKeyIdentifier',
+                                      privateKeyIdentifier);
+    }
+
+    if (typeof privateKeyIdentifier === 'string' &&
+        typeof privateKeyEngine === 'string') {
+      if (c.context.setEngineKey)
+        c.context.setEngineKey(privateKeyIdentifier, privateKeyEngine);
+      else
+        throw new ERR_CRYPTO_CUSTOM_ENGINE_NOT_SUPPORTED();
+    } else if (typeof privateKeyIdentifier !== 'string') {
```

Comment:
This doesn't support `null`, I'm fine with that, but the warning messages below say that `null` is supported type.

Reply:
Removed `null` from the messages.